pkg/runtime/reconciler.go

```diff
@@ -35,27 +35,31 @@ import (
 	acktypes "github.com/aws-controllers-k8s/runtime/pkg/types"
 )
 
-// reconciler is responsible for reconciling the state of a SINGLE KIND of
-// Kubernetes custom resources (CRs) that represent AWS service API resources.
-// It implements the upstream controller-runtime `Reconciler` interface.
-//
-// The upstream controller-runtime.Manager object ends up managing MULTIPLE
-// controller-runtime.Controller objects (each containing a single reconciler
-// object)s and sharing watch and informer queues across those controllers.
+// reconciler describes a generic reconciler within ACK.
 type reconciler struct {
-	sc      *ServiceController
 	kc      client.Client
-	rmf     acktypes.AWSResourceManagerFactory
-	rd      acktypes.AWSResourceDescriptor
 	log     logr.Logger
 	cfg     ackcfg.Config
 	cache   ackrtcache.Caches
 	metrics *ackmetrics.Metrics
 }
 
+// resourceReconciler is responsible for reconciling the state of a SINGLE KIND of
+// Kubernetes custom resources (CRs) that represent AWS service API resources.
+// It implements the upstream controller-runtime `Reconciler` interface.
+//
+// The upstream controller-runtime.Manager object ends up managing MULTIPLE
+// controller-runtime.Controller objects (each containing a single resourceReconciler
+// object)s and sharing watch and informer queues across those controllers.
+type resourceReconciler struct {
+	reconciler
+	rmf acktypes.AWSResourceManagerFactory
+	rd  acktypes.AWSResourceDescriptor
+}
```

Comment:
Nice. This is exactly the kind of polymorphism I was thinking of when we chatted about the implementation. ++